adal/tests/Test.ADAL.NET.UIAutomation/CoreMobileMSALTests.cs

```diff
@@ -0,0 +1,145 @@
+        public static void AcquireTokenADFSvXInteractiveMSALTest(ITestController controller, FederationProvider federationProvider, bool isFederated)
+        {
+            AcquireTokenInteractivly(controller);
+            VerifyResult(controller);
+        }
+
+        private static void AcquireTokenInteractivly(ITestController controller)
+        {
+            var user = prepareForAuthentication(controller);
+            SetInputData(controller, UIAutomationAppV2, DefaultScope);
+            PerformSignInFlow(controller, user);
+        }
+
+        private static IUser prepareForAuthentication(ITestController controller)
+        {
+            //Clear Cache
+            controller.Tap(CachePageID);
+            controller.Tap(ClearCacheID);
+
+            //Get User from Lab
+            return controller.GetUser(
+                new UserQueryParameters
+                {
+                    IsMamUser = false,
+                    IsMfaUser = false,
+                    IsFederatedUser = false
+                });
+        }
+
+        private static void SetInputData(ITestController controller, string ClientID, string scopes)
+        {
+            controller.Tap(SettignsPageID);
+
+            //Enter ClientID
+            controller.EnterText(clientIdEntryID, ClientID, false);
+            controller.DismissKeyboard();
+            controller.Tap(SaveID);
+
+            //Enter Scopes
+            controller.Tap(AcquireTokenID);
+            controller.EnterText(ScopesEntryID, scopes, false);
+            controller.DismissKeyboard();
+        }
+
+        private static void PerformSignInFlow(ITestController controller, IUser user)
+        {
+            string passwordInputID = string.Empty;
+            string signInButtonID = string.Empty;
+
+            switch (user.FederationProvider)
+            {
+                case FederationProvider.AdfsV4:
+                    passwordInputID = AdfsV4WebPasswordID;
+                    signInButtonID = AdfsV4WebSubmitID;
+                    break;
+                default:
+                    passwordInputID = WebPasswordID;
+                    signInButtonID = WebSubmitID;
+                    break;
+            }
+
+            //Acquire token flow
+            controller.Tap(AcquireTokenID);
+            //i0116 = UPN text field on AAD sign in endpoint
+            controller.EnterText(WebUPNInputID, 20, user.Upn, true);
+            //idSIButton9 = Sign in button
+            controller.Tap(WebSubmitID, true);
+            //i0118 = password text field
+            controller.EnterText(passwordInputID, ((LabUser)user).GetPassword(), true);
+            controller.Tap(signInButtonID, true);
+        }
+
+        private static void VerifyResult(ITestController controller)
+        {
+            //Test results are put into a label that is checked for messages
+            Assert.IsTrue(controller.GetText(TestResultID).Contains(TestResultSuccsesfulMessage));
```

Comment:
>Assert.IsTrue(controller.GetText(TestResultID).Contains(TestResultSuccsesfulMessage)); [](start = 12, length = 86)

remember to clear the text field before running tests, otherwise one successful test will result in all tests being successful.

Reply:
This is cleared before every authentication call by the test app itself